modules/apim-adapter/src/main/java/com/axway/apim/adapter/clientApps/APIMgrAppsAdapter.java

```diff
@@ -374,19 +374,24 @@ private void saveCredentials(ClientApplication app, ClientApplication actualApp)
 		HttpResponse httpResponse = null;
 		for(ClientAppCredential cred : app.getCredentials()) {
 			if(actualApp!=null && actualApp.getCredentials().contains(cred)) continue;
+			FilterProvider filter;
 			if(cred instanceof OAuth) {
 				endpoint = "oauth";
+				filter = new SimpleFilterProvider().setDefaultFilter(
+						SimpleBeanPropertyFilter.serializeAllExcept(new String[] {"credentialType", "clientId", "apiKey"}));
 			} else if (cred instanceof ExtClients) {
 				endpoint = "extclients";
+				filter = new SimpleFilterProvider().setDefaultFilter(
+						SimpleBeanPropertyFilter.serializeAllExcept(new String[] {"credentialType", "apiKey"}));
```

Comment:
I have to serialize the clientid in this case

Reply:
Have you tested this with oauth as well? I'm thinking, the clientId needs to be serialized as well for oauth, as it's using the same field. Do you had the change to test this as well?

Reply:
Yes, I tested it, in API Manager web interface what you see in clientId text field for oauth credential is the id field in REST API. Only the /extclients API defines the clientId field from what I saw in API Manager REST API swagger file.

Reply:
Perfect! Thanks for your confirmation. That I was no totally drunk as I have implemented this. 😃